### PR TITLE
Refactor SVG Element Properties for Proper and Improved Usability

### PR DESCRIPTION
**Pull request by binayshaw7777**
### This PR partially resolved: #307 

Added proper documentation as well!
Also resolved raised suggestions in comments and Nit comments. [Previous PR](https://github.com/varabyte/kobweb/pull/314)

SVGs covered so far:
- [x] Circle
- [x] Line
- [x] Rect
- [x] Polyline
- [x] Polygon
- [x] Ellipse
- [ ] Path
- [ ] Group
- [ ] Text

### CODE REVIEW

File: frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/dom/Svg.kt [hunk 1]
```diff
@@ -53,18 +65,91 @@ fun Svg(
...
+
+    fun ry(value: CSSLengthOrPercentageValue) { attrs.attr("ry", value.value.toString()) }
 }
 
-// https://developer.mozilla.org/en-US/docs/Web/SVG/Element/ellipse
+
+/**
+ * Type-safe API for creating an [SVGEllipseElement].
+ *
+ * For example, to create a circle with a radius of 8 centered at 12x12:
```
Comment (bitspittle) [inline frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/dom/Svg.kt:130]: Nit: Update comment here to match code below

File: frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/dom/Svg.kt [hunk 2]
```diff
@@ -53,18 +65,91 @@ fun Svg(
...
+ * Svg {
+ *  Circle {
+ *      cx(12)
+ *      cy(12)
+ *      r(8)
+ *  }
+ * }
+ * ```
+ *
+ * @see <a href="https://developer.mozilla.org/en-US/docs/Web/SVG/Element/circle">SVG Element Circle (Mozilla Docs)</a>
```
Comment (bitspittle) [inline frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/dom/Svg.kt:97]: Wow, I didn't know you could use `@see` in this way. The official docs don't mention it: https://kotlinlang.org/docs/kotlin-doc.html#see-identifier

How did you figure that out?

File: frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/dom/Svg.kt [hunk 3]
```diff
@@ -74,34 +159,181 @@ fun ElementScope<SVGElement>.Group(
...
+    fun ry(value: CSSLengthOrPercentageValue) { attrs.attr("ry", value.value.toString()) }
+}
+
+/**
+ * Type-safe API for creating an [SVGRectElement].
+ *
+ * For example, to create a Rect and Rounded Corner Rect of size 100x100 :
+ *
+ * ```
+ * // Simple Rectangle
```
Comment (bitspittle) [inline frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/dom/Svg.kt:308]: Very nice, I like how you include two different examples here

**Review by bitspittle — APPROVED** [review]
I think this looks great, thank you!

There was only one comment issue I saw (with Ellipse), but that's trivial.

File: frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/dom/Svg.kt (hunk 1 above)
Comment (binayshaw7777) [reply frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/dom/Svg.kt:130]: Updated.

File: frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/dom/Svg.kt (hunk 2 above)
Comment (binayshaw7777) [reply frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/dom/Svg.kt:97]: I just came across this article: [Here](https://www.baeldung.com/javadoc-linking-external-url)  😄

**Pull request merged**